### PR TITLE
add realesrgan to requirements.txt, remove nighties for torch and torchvision

### PR DESCRIPTION
**Pull request by Vargol**
Add realesrgan to requirements.txt, remove nighties for torch and torchvision due to performance issues

### CODE REVIEW

**Comment by Any-Winter-4079** [comment]
Can someone on CUDA give the green light (since this affects `requirements.txt` which I assume is shared between CUDA and MPS)?

**Comment by Any-Winter-4079** [comment]
@Vargol The `nightly` change looks good.
Also `--pre` (to include pre-release and development versions, as by default, pip only finds stable versions) was removed, but I assume it only affected nightly?

Oh, another thing. Why are some packages not included, such as `coloredlogs`, not included in requirements.txt? I'm sure there are more instances too of packages on environment files not present on `requirements.txt`

**Comment by Vargol** [comment]
Removing pre should only affect the pytorch nightlies.
None of the environment/requirements files other than `environment-mac.yml` have `coloredlog` in, 
it may an old obsolete requirement,

A search in the repo via the Github page search textbox does not find a code reference. I just made the minimum changes to the files to get it working with a pure pip install.

I've also run the brute force search

(req_fix_invoke_ai) % find . -type f -exec grep -i coloredlogs {} /dev/null \;
./environment-mac.yml:  - coloredlogs==15.0.1
(req_fix_invoke_ai)  % 

The example in PyPy shows in imported as
`import coloredlogs, logging`

so the brute force search should have found it, if it was required.

**Comment by lstein** [comment]
If you folks are satisfied with this then I'll merge it in.

**Comment by Any-Winter-4079** [comment]
@Vargol `coloredlog` is for terminal output using different colors. I'm pretty sure at some point I've seen those colors. Maybe generating variations? Maybe they've removed them since, though.

**Comment by Vargol** [comment]
I've seen the coloured traceback output, but I believe that come from rich which comes in as a dependancy for something else

```
 % pip list | grep rich
rich                        12.6.0
```

File: requirements.txt [hunk 1]
```diff
@@ -31,6 +31,7 @@ flaskwebgui==0.3.7
 send2trash
 dependency_injector==4.40.0
 eventlet
+realesrgan
```
Comment (tildebyte) [inline requirements.txt:34]: @Vargol; Remind of why we're making this a hard requirement for *everyone*, regardless of whether they use esrgan or not?

**Comment by tildebyte** [comment]
@Any-Winter-4079;

> requirements.txt which I assume is shared between CUDA and MPS

It is shared between ALL pip installs, on ALL platforms

> If requirements.txt is for non-conda setups, maybe it'd be more obvious to call it requirements-no-conda.txt. So that people using conda know not to use it, and vice versa. I find the name a bit ambiguous right now.

*"environment.yaml" is for NON-PIP installs*. "requirements.txt" is THE standard name which pip looks for. As intimated above, ours is the master for pip installs, which contains all common dependencies.

File: requirements.txt (hunk 1 above)
Comment (Vargol) [reply requirements.txt:34]: It's is the tests we do as part of the pre-flight testing, its part of embiggen,  preload-models.py doesn't fetch the GPFGAN models it its not installed.
Users will forget they didn't install it and then wonder why that prompt they copied doesn't work especially then they see that bit of dream.py saying its been initialised.

**Comment by Any-Winter-4079** [comment]
@tildebyte so basically for people that don't use conda and want to install everything via pip, right?
Because I use conda and simply use `environment-mac.yaml` which has pip dependencies already in it.

File: requirements.txt (hunk 1 above)
Comment (tildebyte) [reply requirements.txt:34]: I take your point(s), but the conda and pip installs are not equivalent (conda does weird stuff like specify "pytorch", which does not exist), so please *do not* base decisions about changes to either, given the state of the other.

**Comment by tildebyte** [comment]
> @tildebyte so basically for people that don't use conda and want to install everything via pip, right? Because I use conda and simply use `environment-mac.yaml` which has pip dependencies already in it.

To a first approximation , yes, but I'm trying to point out that pip is an actual **part of** Python, it's in the standard library, it is the default mecchanism for package installs, etc. whereas conda is it's own thing, which wraps pip (partially - it does lots of other things as well).

TL;DR pip is the standard install mechanism in the Python world - not conda. And given how much trouble I've seen with conda installs in this repo[1], I find that a significant fact.

[1] I am in no way claiming that pip installs never fail, only that pip is 1) completely transparent (a literal part of Python) 2) very well understood. I wouldn't say either of those things about conda.

**Comment by Any-Winter-4079** [comment]
@tildebyte Yeah, I meant for this project, as an easy way to think about it.

* people using conda -> `environment.yml` (or `environment-mac.yml`) has everything they need.
* people not using conda -> ditto for `requirements.txt` (or machine-specific version, e.g. `requirements-mac-MPS-CPU.txt` which in this case uses `requirements.txt`, although in other cases such as `requirements-linux-arm64.txt` it is self-contained).

Of course, the pip dependencies included in the `environment yml` files will probably match a subset of those in `requirements` file, so they are certainly interconnected.

Oh and thanks for the explanation! <3

**Comment by tildebyte** [comment]
You're certainly welcome!

> Of course, the pip dependencies included in the environment yml files will probably match a subset of those in requirements file, so they are certainly interconnected.

Definitely. The major practical difference in that sense is that conda handles platform-specific binary packages *very* differently than straight pip does - thus the bizarre "pytorch" reference, which apparently is a specific conda "channel".
